### PR TITLE
Update CONTRIBUTING.md & fix file ownership

### PR DESCRIPTION
**Pull request by dagardner-nv**
Largely this is needed because both the conda docker build and the morpheus build will share the `.cache/ccache` dir

### CODE REVIEW

**Comment by GPUtester** [comment]
Can one of the admins verify this patch?

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -151,40 +155,62 @@ Note: These instructions assume the user is using `mamba` instead of `conda` sin
...
+1. Test the build (Note: some tests will be skipped)
+   ```bash
+   pytest
+   ```
+1. Optional: Run full end-to-end tests
+   - Our end-to-end tests require the [camouflage](https://testinggospels.github.io/camouflage/) testing framework. Install camouflage with:
+      ```bash
+      npm install -g camouflage-server
+      ```
+
```
Comment (mdemoret-nv) [inline CONTRIBUTING.md:208]: Note that this also requires nodejs to be installed

File: docker/build_conda_packages.sh [hunk 2]
```diff
@@ -85,6 +90,7 @@ echo "Running conda build"
 
 # Run with an output folder that is mounted and skip existing to avoid repeated builds
 DOCKER_EXTRA_ARGS="${DOCKER_EXTRA_ARGS[@]}" ${SCRIPT_DIR}/run_container_dev.sh bash -c "${BUILD_SCRIPT}"
+DOCKER_EXTRA_ARGS="${DOCKER_EXTRA_ARGS[@]}" ${SCRIPT_DIR}/run_container_dev.sh bash -c "chown -R ${CUR_UID}:${CUR_GID} .cache"
```
Comment (mdemoret-nv) [inline docker/build_conda_packages.sh:93]: Can these be combined into a single command?

**Review by mdemoret-nv — CHANGES_REQUESTED** [review]

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -151,40 +155,62 @@ Note: These instructions assume the user is using `mamba` instead of `conda` sin
...
    ```
-   This creates an environment named `morpheus` with all necessary dependencies, and activates that environment.
-2. Build cuDF
+
+   This creates a new environment named `morpheus`, and activates that environment.
+1. Set up `ssh-agent` to allow container to pull from private repos
+   ```bash
+   eval `ssh-agent -s`
+   ssh-add
+   ```
```
[the remark this answers is not in the document]
Comment (dagardner-nv) [reply CONTRIBUTING.md:178]: I believe the issue is that without it the `./docker/build_conda_packages.sh` command will fail without it. Alternately I could update that script to run `ssh-agent -s` if `SSH_AUTH_SOCK` is undefined.

[the remark this answers is not in the document]
Comment (mdemoret-nv) [reply CONTRIBUTING.md:178]: Why does `./docker/build_conda_packages.sh` fail without it?

File: CONTRIBUTING.md (hunk 1 above)
Comment (dagardner-nv) [reply CONTRIBUTING.md:208]: I put it back in `docker/conda/environments/cuda11.4_dev.yml` in this PR

File: docker/build_conda_packages.sh (hunk 2 above)
Comment (dagardner-nv) [reply docker/build_conda_packages.sh]: done

**Review by mdemoret-nv — APPROVED** [review]

**Pull request merged**